src/psyclone/psyir/nodes/loop.py

```diff
@@ -111,9 +113,42 @@ def __init__(self, parent=None, variable_name="", valid_loop_types=None,
         # TODO replace iterates_over with iteration_space
         self._iterates_over = "unknown"
 
-        self._variable_name = variable_name
+        self._check_variable(variable)
+        self._variable = variable
         self._id = ""
 
+    @staticmethod
+    def _check_variable(variable):
+        '''The loop variable should be a scalar integer. Check that this is
+        the case and raise an exception if not.
+
+        :param variable: the loop iterator.
+        :type variable: :py:class:`psyclone.psyir.symbols.DataSymbol`
+
+        :raises GenerationError: if the supplied variable is not a \
+            scalar integer.
+
+        '''
+        if not variable:
+            # No variable has been supplied so there is nothing to
+            # check
+            return
+        if not isinstance(variable, DataSymbol):
+            raise GenerationError(
+                "variable argument in create method of Loop class should "
```

Comment:
`_check_variable` is also called from the constructor and the variable setter (next comment), but the three error messages in this method point to just he create method. Can you make the 3 error message more generic? I wouldn't include the method name as it is already visible in the error stack trace.

Reply:
Good spot. I've modified the text.